src/app/utils/query-url-sanitization.ts

```diff
@@ -1,67 +1,74 @@
 const quotedTextRegex = /"([^"]*)"/g;
 
-// Matches PII patterns
-const numberRegex =  /(?<=\/)\d+\b/g; // number between forward slashes
-const emailRegex = /([a-z0-9_\-.+]+)@\w+(\.\w+)*/gi;
-const guidRegex =
- /(\{){0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}/gi;
+// matches strings that are all alphabets
+const allAlphaRegex = /^[A-Za-z]+$/;
 
+// matches strings with depracation identifier
+const depracationRegex = /_v2/gi;
 
 export function sanitizeQueryUrl(url: string): string {
   url = decodeURIComponent(url);
 
   // Extract query string
-  const { search } = parseSampleUrl(url);
+  const { search, queryVersion, requestUrl } = parseSampleUrl(url);
   const queryString: string = sanitizeQueryParameters(search);
 
-  // Drop query string
-  let sanitizedUrl = url.split('?')[0];
-
-  // Normalize parameters in param=<arbitraryKey> format with param={value} format
-  sanitizedUrl = sanitizedUrl.replace(functionParamInitialRegex, '<value>');
-  sanitizedUrl = sanitizedUrl.replace(functionParamFinalRegex, '<value>');
+  let resourceUrl = requestUrl;
+  const sections = requestUrl.split('/');
+  sections.forEach(segment => {
+    if (isAllAlpha(segment) || isDeprecation(segment)) {
+      return;
+    }
 
-  // Replace IDs and ID placeholders with generic {id}
-  sanitizedUrl = sanitizedUrl.replace(keyIdRegex, '/<id>');
-  sanitizedUrl = sanitizedUrl.replace(guidRegex, '<guid>');
-  sanitizedUrl = sanitizedUrl.replace(numberRegex, '<number>');
+    const index = sections.indexOf(segment);
+    const replacementItemWithPrefix = `{${sections[index - 1]}-id}`;
+    resourceUrl = resourceUrl.replace(segment, replacementItemWithPrefix);
+});
 
-  // Redact PII
-  sanitizedUrl = sanitizedUrl.replace(emailRegex, '<email>');
-  return `${sanitizedUrl}${queryString}` ;
+  return `${GRAPH_URL}/${queryVersion}/${resourceUrl}${queryString}`;
 }
 
 function sanitizeQueryParameters(queryString: string): string {
-   const params = queryString.split('&');
-   let result: string = '';
-   if (params.length)
-   {
-      params.forEach(param => {
-       result += sanitizeQueryParameterValue(param) + '&';
-      });
-      result = result.slice(0, -1);
-   }
-   return result;
+  const params = queryString.split('&');
+  let result: string = '';
+  if (params.length) {
+    params.forEach(param => {
+      result += sanitizeQueryParameterValue(param) + '&';
+    });
+    result = result.slice(0, -1);
+  }
+  return result;
+}
+
+/**
+ * @param segment part of the url string to test
+ * Currently, non-ID strings are all alphabetic characters
+ * @returns boolean
+ */
+export function isAllAlpha(segment: string): boolean {
+  return !!segment.match(allAlphaRegex);
+}
+
+/**
+ * @param segment part of the url string to test
+ * depracated resources may have `_v2` temporarily
+ * @returns boolean
+ */
+export function isDeprecation(segment: string): boolean {
```

Comment:
I have already signed off on this, but an edge case woke me up from my sleep last night :D

- An email can contain `_v2` and we shouldn't mark it as Non-ID in that case. We can either add a new condition here that the string doesn't include `@` symbol in addition to containing `_v2`. Or we can improve the regular expression for the deprecation regex to always start with only alpha characters before `_v2`, something like: `^[A-Za-z]+_v2$`